### PR TITLE
Added parameter expansion.

### PR DESCRIPTION
**Pull request by digitalBush**

### CODE REVIEW

File: spec/integration/basic.spec.js [hunk 1]
```diff
@@ -98,8 +98,9 @@ describe( "Basic - Integration", () => {
 
 		it( "should work with a value array param", () => {
 			const query = `
-				SELECT value
-				FROM @userIds`;
+				SELECT * from (select 1 id union select 2 union select 3 ) t
```
Comment (calvinb) [inline spec/integration/basic.spec.js:101]: Look, Ma; no tables! 😂

File: src/parameterBuilder.js [hunk 2]
```diff
@@ -48,14 +70,13 @@ module.exports = function( request, params ) {
 		if ( Array.isArray( param.val ) ) {
```
Comment (calvinb) [inline src/parameterBuilder.js]: You could eliminate some nesting by inverting this condition.

File: src/parameterBuilder.js [hunk 3]
```diff
@@ -13,26 +13,48 @@ function typeDeclarations( definition ) {
...
+	if ( param.val.length === 0 ) {
+		request.sqlTextOrProcedure = request.sqlTextOrProcedure.replace( matcher, "(SELECT 1 WHERE 1=0)" );
+		return;
+	}
+
+	const { type: { type, length, precision, scale } } = param;
+
+	const params = [];
+	let i = 0;
+	for ( const item of param.val ) {
```
Comment (calvinb) [inline src/parameterBuilder.js:27]: I can't help trying to make this loop less imperative. I might be reaching too hard here, but what if you set `params` to a `map` over `param.val`? That would save you from declaring and incrementing `i`. Then you could still call `request.addParameter` in the mapped function.

Comment (digitalBush) [reply src/parameterBuilder.js:27]: Feels wrong to have a map with side effects though.

File: src/parameterBuilder.js (hunk 2 above)
Comment (digitalBush) [reply src/parameterBuilder.js]: Swapped this up a bit. What do you think?

File: src/parameterBuilder.js (hunk 3 above)
Comment (calvinb) [reply src/parameterBuilder.js:27]: I thought you might say that! And I agree. My only other idea was a reduce with the request and params in a tuple as the accumulator.

**Pull request merged**